enhancements/hybrid-helm-operator.md

```diff
@@ -0,0 +1,223 @@
+  it is essentially a Go-based operator and so will not make use of a base image with
+  dependencies, automatic upgrades of helm versions, or any other conveniences provided
+  by the base image. The user will need to manage and upgrade their code by hand.
+
+## Proposal
+
+There are 3 stages to this proposal:
+
+1. Joe Lanford has made significant progress on this already, in the
+   [joelanford/helm-operator][helm-operator-repo] repository.
+   First we will migrate this code to the operator-sdk repo, where it will replace the
+   existing helm-operator code.
+1. We will write a new SDK plugin that can scaffold out a hybrid operator. As a first
+   iteration we can just scaffold out most of the code that is currently in the
+   [helm-operator][helm-operator-repo] (though the exact structure of the code will likely change)
+   and use that to generate what we ship in the base image.
+1. We will break as much code as we reasonably can out of the scaffolding and into a
+   utility library, to minimize the amount of code the user will need to maintain. This
+   library could either be the existing operator-lib package, or a new, helm-operator specific
+   package.
+
+### User Stories
+
+#### Story 1
+
+As a Helm-based operator author, I want to add additional programmatic backup and restore
+functionality to my operator, without starting over with a completely different type of operator.
+
+#### Story 2
+
+As a Helm-based operator author, I want to change core behavior of the reconciliation loop
+that is not currently exposed in the watches.yaml
+
+#### Story 3
+
+As a prospective author of a Helm-based operator, I want to know that if I invest time and
+effort creating a Helm-based operator, I won't need to start over in order to create a higher
+level operator.
+
+#### Story 4
+
+As a Golang-based operator author, I would like to leverage existing helm charts for deploying
+pieces of my application, without sacrificing the flexibility of Go.
+
+
+### Implementation Details/Notes/Constraints
+
+#### Scaffolding plugin for the SDK
+
+The basic idea behind the hybrid Helm/Go operator plugin is that we will scaffold out a Go
+operator project structure, with all of the code bits from the helm-operator project filled in,
+such that building the project without adding any APIs or modifying any code will result in the
+same container as what is shipped in the helm-operator base image.
+
+Because the driving need behind this functionality is the ability to customize any piece of
+the Helm-based operator, we will scaffold out every bit of code that is not accessible via
+a library, and err on the side of scaffolding more than we need to.
+
+For the first iteration of the hybrid Helm/Go operator plugin, we will likely just scaffold
+out nearly everything that is currently in the [helm-operator repository][helm-operator-repo], but
+with a `main.go` that is more in line with a typical Go-based operator, including the kubebuilder
```

Comment:
can we make this way, when we do `operator-sdk init` we scaffold a normal go based operator. When we do `operator-sdk create api --helm-chart <xxx>` then we scaffold `chart_types.go` for this (its a separate topic by itself, since we need to figure out how to do this) and the controller. And only when someone does `operator-sdk create api --group <xxx> --version <xxx> --kind <xxx> --go-api` then we scaffold the go api type.

Reply:
I should probably go into more detail here, I was imagining there would be a new Go/Helm hybrid plugin, so it would look like:

```
operator-sdk init --plugins hybrid #name tbd, this would scaffold out main.go as well as an empty watches.yaml
operator-sdk --plugins go create api ...# This would probably also be the default plugin if the --plugins argument is not included
operator-sdk --plugins helm create api ...
```
so you would literally  be using the other plugins to scaffold code after initialization, I don't think we'd necessarily even implement `create api` for the hybrid plugin

Reply:
The user would specify the plugin to use with the --plugins flag I think (at least at first), so it would be using the other plugins directly